File: java/src/main/java/io/fivexer/sdk/model/WorkerLocaleState.java
package io.fivexer.sdk.model;

/**
 * A worker's own language, after setting it.
 *
 * <p>Populated by Gson via field reflection and exposed read-only; callers receive instances
 * from the client and never construct them.
 */
public class WorkerLocaleState {
    private String workerId;
    private String locale;

    public String getWorkerId() { return workerId; }
    public String getLocale() { return locale; }
}

File: php/src/Model/WorkerLocaleState.php
<?php

declare(strict_types=1);

namespace Fivexer\SDK\Model;

/**
 * A worker's own language, after setting it.
 *
 * Response model: built from a decoded JSON array and read-only thereafter.
 */
final class WorkerLocaleState
{
    public function __construct(
        public readonly string $workerId,
        public readonly ?string $locale,
    ) {
    }

    /**
     * @param array<string, mixed> $data
     * @internal
     */
    public static function fromArray(array $data): self
    {
        return new self(
            workerId: (string) ($data['workerId'] ?? ''),
            locale: isset($data['locale']) ? (string) $data['locale'] : null,
        );
    }
}
